### PR TITLE
FSE: Merge Blog Posts block with latest from newspacks-blocks

### LINKED ISSUE #39251: Homepage Articles block: allow multiple instances per post
As reported on newspack-blocks repo (https://github.com/Automattic/newspack-blocks/issues/317), adding multiple Homepage Articles (aka Blog Posts) blocks should be possible. 

This was done deliberately in https://github.com/Automattic/wp-calypso/issues/38404:
https://github.com/Automattic/wp-calypso/blob/b6e7b09b1be3c9b8eb9d790755ec8c9adb1cd23f/apps/full-site-editing/full-site-editing-plugin/blog-posts-block/editor.js#L36
This issue is more of a reminder and a place for further discussion.

As @kspilarski mentioned, this surfaced in 2666978-zd

### PR DESCRIPTION
**Pull request by obenland**
Updates FSE with the latest Blog Posts Blog changes.

Fixes #39251, #39212, #39509.

### CODE REVIEW

File: apps/full-site-editing/full-site-editing-plugin/starter-page-templates/class-starter-page-templates.php [hunk 1]
```diff
@@ -212,7 +212,7 @@ public function fetch_vertical_data() {
 			$request_url = add_query_arg(
 				[ '_locale' => $this->get_iso_639_locale() ],
 				'https://public-api.wordpress.com/wpcom/v2/verticals/' . $vertical_id . '/templates'
-			);
+			);error_log($request_url);
```
Comment (noahtallen) [inline apps/full-site-editing/full-site-editing-plugin/starter-page-templates/class-starter-page-templates.php]: Was this accidentally included? :)

Comment (obenland) [reply apps/full-site-editing/full-site-editing-plugin/starter-page-templates/class-starter-page-templates.php]: Yes. I reverted it yesterday but forgot to push

**Comment by retrofox** [comment]
> Updates FSE with the latest Blog Posts Blog changes.

what was the release of the Newspack block in the sync process?

**Comment by retrofox** [comment]
According to some slack discussion, we agree that we should use a stable version instead of `master` branch.

**Comment by noahtallen** [comment]
The current version on dotcom is `1.0.0-alpha.20`, but the latest version is [`1.0.0-alpha.25`](https://github.com/Automattic/newspack-blocks/releases).

If you want, we can wait for #39196 to land which would reference the package via npm instead of including the files in source control. (But that might not land until next week, so it depends how soon we want the update.)

**Comment by retrofox** [comment]
Also, we are currently working on a fix for https://github.com/Automattic/newspack-blocks/issues/306. Maybe it worths try to get it in the next FSE release, too.

**Pull request merged**